packages/nodes/emoji/src/createEmojiPlugin.ts

```diff
@@ -0,0 +1,42 @@
+import { createPluginFactory } from '@udecode/plate-core';
+import { EmojiCombobox } from '@udecode/plate-ui-emoji';
+import { ELEMENT_EMOJI, ELEMENT_EMOJI_INPUT, TRIGGER } from './constants';
+import { getOnKeyDownEmoji } from './handlers';
+import { EmojiComboboxProps, EmojiPlugin } from './types';
+import { EmojiTriggeringController } from './utils';
+import { withEmoji } from './withEmoji';
+
+export const createEmojiPlugin = createPluginFactory<EmojiPlugin>({
+  key: ELEMENT_EMOJI,
+  isElement: true,
+  isInline: true,
+  isVoid: true,
+  handlers: {
+    onKeyDown: getOnKeyDownEmoji(),
+  },
+  withOverrides: withEmoji,
+  options: {
+    trigger: TRIGGER,
+    createEmoji: (item) => item.data.emoji,
+    emojiTriggeringController: new EmojiTriggeringController(TRIGGER),
+  },
+  plugins: [
+    {
+      key: ELEMENT_EMOJI_INPUT,
+      isElement: true,
+      isInline: true,
+    },
+  ],
+  then: (
+    _,
+    { key, options: { trigger, createEmoji, emojiTriggeringController } }
+  ) => ({
+    options: {
+      id: key,
+      trigger,
+      createEmoji,
+      emojiTriggeringController,
+    },
+  }),
+  renderAfterEditable: EmojiCombobox as EmojiComboboxProps,
```

Comment:
We can't import `@udecode/plate-ui-emoji'` here, see next comment...

Reply:
updated